src/init/cli/CliExtractor.ts

```diff
@@ -0,0 +1,14 @@
+import { AsyncHandler } from '../../util/handlers/AsyncHandler';
+
+/**
+ * Converts the input CLI arguments into an easily parseable key/value object.
+ *
+ * There are certain CLI parameters that are required before this class can be instantiated.
+ * These can be ignored by this class, but that does mean that this class should not error if they are present.
+ *
+ * Those CLI parameters are specifically:
```

Comment:
> And why not reflect that in the typing?

The input is the input argv string array, so no real way to type that. It's also not even guaranteed they are in there and in what form. It could be `-c config`, `--config config` or neither in case of the default config.